File: src/main/java/com/timurisachenko/websocketchat/model/ChatMessage.java
package com.timurisachenko.websocketchat.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class ChatMessage {
    private MessageType type;
    private String roomId;
    private String from;
    private String message;
    private long userCount;
    private LocalDateTime timestamp;
    private MessageStatus status;

    public ChatMessage() {
    }

    @Builder
    public ChatMessage(MessageType type, String roomId, String from, String message, long userCount) {
        this.type = type;
        this.roomId = roomId;
        this.from = from;
        this.message = message;
        this.userCount = userCount;
    }

    public enum MessageType {
        ENTER, TALK, QUIT
    }
}

File: src/main/kotlin/com/itime/sernachat/domain/ChatMessage.kt
package com.itime.sernachat.domain

import java.time.LocalDateTime

class ChatMessage private constructor(
    var type: MessageType?,
    var roomId: String?,
    var from: String?,
    var message: String?,
    var userCount: Long? = 0
) {

    enum class MessageType {
        ENTER, TALK, QUIT
    }

    data class Builder(
        var type: MessageType? = null,
        var roomId: String? = null,
        var from: String? = null,
        var message: String? = null,
        var userCount: Long? = 0
    ) {

        fun type(type: MessageType) = apply { this.type = type }
        fun roomId(roomId: String) = apply { this.roomId = roomId }
        fun from(from: String) = apply { this.from = from }
        fun message(message: String) = apply { this.message = message }
        fun build() = ChatMessage(type, roomId, from, message, userCount)
    }
}